src/App.jsx

```diff
@@ -1,18 +1,62 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import API from "./API";
 import Echo from "./Echo";
-import Login from "./Login";
 import Registration from "./Registration";
+import Hub from "./pages/Hub";
+import RegPage from "./pages/RegPage";
+import LoginPage from "./pages/LoginPage";
+import LogoutPage from "./pages/LogoutPage";
+import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
+import 'bootstrap/dist/css/bootstrap.min.css';
+import './styles/App.css';
+import SettingsHeader from "./views/SettingsHeader";
+import LoginProvider from "./util/LoginProvider";
 
 export default function () {
-  return (
-    <>
-      <section>
-        <span>Using endpoint: {API.baseUrl}</span>
-      </section>
-      <Registration />
-      {/* <Login /> */}
-      <Echo />
-    </>
-  );
+    const [cluster, setCluster] = useState("live");
+    const [flow, setFlow] = useState("browser");
+    const [sessionActive, setSessionActive] = useState(false);
+
+    const clusterChange = (event) => {
+        setCluster(event.target.value);
+    }
+
+    const flowChange = (event) => {
+        setFlow(event.target.value);
+    }
+
+    const checkSession = async () => {
+        let login = new LoginProvider(cluster, flow);
+        var jsonRes = await API.makeRequest(login.getSessionRequest());
+
+        if (jsonRes.status == 401) {
+            setSessionActive(false);
+        } else {
+            setSessionActive(true);
+        }
+    }
+
+    useEffect(() => {
+        checkSession();
+    }, []);
```

Comment:
```suggestion
    useEffect(checkSession, [flow, cluster]);
```
By passing the empty array, you are circumventing the issue caused by called a `setState` hook in a `setEffect` hook. However, you will not rerender even if that would be desirable. We therefore add `flow` and `cluster` as effect triggers.

Reply:
Why does this trigger a re-render even if flow and cluster are not changed?